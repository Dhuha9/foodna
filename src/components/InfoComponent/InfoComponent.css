@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

* {
    box-sizing: border-box;
}

.gray-theme {
    background-color:#F2F2F2;
    color:#000;
}

.gray-theme .info-img {
    box-shadow: -50px -50px white;

}

.green-theme {
    background-color:#03B664;
    color:#fff;
}

.green-theme .info-img {
    box-shadow: -50px -50px white;

}

@media screen and (min-width: 600px) {
    .green-theme {
      display: table;
      width: 100%;
      direction: rtl;
    }
   
    .green-theme content {
      display: table-cell;
      width: 25%;
    }
}

.white-theme {
    background-color:#fff;
    color:#000;
}

.white-theme .info-img {
    box-shadow: -50px -50px #E5E5E5;

}

.content, .media{
    width: 50%;
    display: inline-block;
}

.content{
    padding: 150px;
}

.content h2{
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 32px;
    line-height: 37px;
}

.content p{
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 25px;
}

.content p.small{
    font-family:'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 20px;
    color: rgba(0, 0, 0, 0.5);
}

.media{
    text-align: center;
}

.info-img {
    width: 400px;
}


@media only screen and (max-width: 700px) {
    /* For mobile screens */
    .content, .media{
        width: 100%;
    }

    .content{
        padding: 0px;
    }

    .info-img {
        width: 200px;
        box-shadow: none;
        margin-top: 20px;
    }
}

@media only screen and (max-width: 1000px) {
    /* For Tablet screens */
    .content, .media{
        width: 100%;
    }

    .content{
        padding: 90px;
    }

    .info-img {
        margin-top: 90px;
    }

}