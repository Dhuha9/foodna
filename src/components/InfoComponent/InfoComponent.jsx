// Use those lines when importing this component in app.js 


// <InfoComponent color={'gray'}/>
// <InfoComponent color={'green'}/>
// <InfoComponent color={'white'}/>


import React from 'react'
import PropTypes from 'prop-types';
import './InfoComponent.css';

const theme = {
    gray:{
        class:'gray-theme',
        img:'img/info1.jpg'
    },
    green:{
        class:'green-theme',
        img:'img/info2.jpg'
    },
    white:{
        class:'white-theme',
        img:'img/info3.jpg'
    }
}

export default function InfoComponent({color}) {
    return (
        <div className={theme[color].class}>
            <div className="media" >
                <img className="info-img" src={theme[color].img} alt=""/>
            </div>
            <div className="content">
                <h2>More info about us</h2>
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
                <p className="small">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
            </div>
        </div>
    )
}

InfoComponent.propTypes = {
    color: PropTypes.string.isRequired
};

