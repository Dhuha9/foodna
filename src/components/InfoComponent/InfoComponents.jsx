// Use those lines when importing this component in app.js

// <InfoComponent color={'gray'}/>
// <InfoComponent color={'green'}/>
// <InfoComponent color={'white'}/>

import React from 'react';
import InfoComponent from '../../components/InfoComponent/InfoComponent';
import { useTranslation } from 'react-i18next';

const themes = [
  {
    class: 'gray-theme',
    img: 'img/info1.jpg',
  },
  {
    class: 'green-theme',
    img: 'img/info2.jpg',
  },
  {
    class: 'white-theme',
    img: 'img/info3.jpg',
  },
];

export default function InfoComponents() {
  const { t } = useTranslation();

  return (
    <div>
      {themes.map((theme, index) => (
        <InfoComponent
          theme={theme}
          key={index}
          title={t(`InfoComponent.Info${index}.title`)}
          content={t(`InfoComponent.Info${index}.content`)}
          subTitle={t(`InfoComponent.Info${index}.subTitle`)}
        />
      ))}
    </div>
  );
}
