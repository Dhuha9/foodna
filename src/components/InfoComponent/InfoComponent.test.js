import React from 'react';
import renderer from 'react-test-renderer';
import InfoComponent from './InfoComponent';

it ("renders when there are is no gray InfoComponent", () => {
    const tree = renderer.create(
        <InfoComponent color='gray'/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it ("renders when there are is no green InfoComponent" , () => {
    const tree = renderer.create(
        <InfoComponent color='green'/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it ("renders when there are is no white InfoComponent", () => {
    const tree = renderer.create(
        <InfoComponent color='white'/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});