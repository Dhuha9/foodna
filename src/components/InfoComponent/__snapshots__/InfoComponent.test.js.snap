// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are is no gray InfoComponent 1`] = `
<div
  className="gray-theme"
>
  <div
    className="media"
  >
    <img
      alt=""
      className="info-img"
      src="img/info1.jpg"
    />
  </div>
  <div
    className="content"
  >
    <h2>
      More info about us
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
    <p
      className="small"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
  </div>
</div>
`;

exports[`renders when there are is no green InfoComponent 1`] = `
<div
  className="green-theme"
>
  <div
    className="media"
  >
    <img
      alt=""
      className="info-img"
      src="img/info2.jpg"
    />
  </div>
  <div
    className="content"
  >
    <h2>
      More info about us
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
    <p
      className="small"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
  </div>
</div>
`;

exports[`renders when there are is no white InfoComponent 1`] = `
<div
  className="white-theme"
>
  <div
    className="media"
  >
    <img
      alt=""
      className="info-img"
      src="img/info3.jpg"
    />
  </div>
  <div
    className="content"
  >
    <h2>
      More info about us
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
    <p
      className="small"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
  </div>
</div>
`;
